OptimizelySDK/Optimizely.cs

```diff
@@ -293,23 +299,16 @@ public void Track(string eventKey, string userId, UserAttributes userAttributes
                 eventTags = eventTags.FilterNullValues(Logger);
             }
 
-            var conversionEvent = EventBuilder.CreateConversionEvent(config, eventKey,
-                userId, userAttributes, eventTags);
+            var userEvent = UserEventFactory.CreateConversionEvent(config, eventKey, userId, userAttributes, eventTags);
+            EventProcessor.Process(userEvent);
             Logger.Log(LogLevel.INFO, string.Format("Tracking event {0} for user {1}.", eventKey, userId));
-            Logger.Log(LogLevel.DEBUG, string.Format("Dispatching conversion event to URL {0} with params {1}.",
-                conversionEvent.Url, conversionEvent.GetParamsAsJson()));
 
-            try
-            {
-                EventDispatcher.DispatchEvent(conversionEvent);
-            }
-            catch (Exception exception)
+            if (NotificationCenter.GetNotificationCount(NotificationCenter.NotificationType.Track) > 0)
             {
-                Logger.Log(LogLevel.ERROR, string.Format("Unable to dispatch conversion event. Error {0}", exception.Message));
-            }
-
-            NotificationCenter.SendNotifications(NotificationCenter.NotificationType.Track, eventKey, userId,
+                var conversionEvent = EventFactory.CreateLogEvent(userEvent, Logger);
```

Comment:
Hmm if we pass it a logger, wouldn't it be weird that the same message gets logged twice because we also create a log event inside the processor. Why not have the forwarding event processor also take in the notification center and send the notification for the notification events? Seems less duplicative than creating the log events multiple times